packages/investor-idle/src/IdleOpportunity.ts

```diff
@@ -327,23 +293,24 @@ export class IdleOpportunity
     // In order to properly earn affiliate revenue, we must deposit to the vault through the SS
     // router contract. This is not necessary for withdraws. We can withdraw directly from the vault
     // without affecting the DAOs affiliate revenue.
-    const tokenChecksum = this.#internals.web3.utils.toChecksumAddress(
-      this.metadata.underlyingAddress
-    )
-    const userChecksum = this.#internals.web3.utils.toChecksumAddress(address)
-    const vaultIndex = await this.getVaultId({
-      underlyingAssetAddress: this.metadata.underlyingAddress,
-      vaultAddress: this.metadata.cdoAddress || this.metadata.address
-    })
-
-    // console.log('prepareDeposit - getVaultId', vaultIndex)
-
-    const preDeposit = await this.#internals.routerContract.methods.deposit(
-      tokenChecksum,
-      userChecksum,
-      amount.toFixed(),
-      vaultIndex
-    )
+
+    let methodName: string
+    let methodParams: string[]
+    let vaultContract: Contract
+
+    // Handle Tranche Withdraw
+    if (this.metadata.cdoAddress) {
+      vaultContract = this.#internals.routerContract
+      const trancheType = this.metadata.strategy.match(/senior/i) ? 'AA' : 'BB'
```

Comment:
Non-blocking but `String.prototype.match()` returns an array, if you just need to test, you can use `String.prototype.test()` which incidentally works but isn't optimal

```suggestion
      const trancheType = /senior/i.test(this.metadata.strategy) ? 'AA' : 'BB'
```